paasta_tools/kubernetes_tools.py

```diff
@@ -3502,14 +3502,19 @@ def get_kubernetes_secret(
     secret_name: str,
     service_name: str,
     namespace: str = "paasta",
+    decode: bool = True,
 ) -> str:
 
     k8s_secret_name = get_kubernetes_secret_name(service_name, secret_name)
 
     secret_data = kube_client.core.read_namespaced_secret(
         name=k8s_secret_name, namespace=namespace
     ).data[secret_name]
-    secret = base64.b64decode(secret_data).decode("utf-8")
+    secret = base64.b64decode(secret_data)
+    # String secrets (e.g. yaml config files) need to be decoded
+    # Binary secrets (e.g. TLS Keystore or binary certificate files) cannot be decoded
```

Comment:
oh interesting: i guess no one has tried storing binary secrets yet and that's why this hasn't popped up before?

Reply:
Yeah I think we just never had this scenario, since historically most paasta secrets end up as environment variable text